github/Project.py

```diff
@@ -0,0 +1,209 @@
+        return self._html_url.value
+
+    @property
+    def id(self):
+        """
+        :type: integer
+        """
+        self._completeIfNotSet(self._id)
+        return self._id.value
+
+    @property
+    def name(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._name)
+        return self._name.value
+
+    @property
+    def node_id(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._node_id)
+        return self._node_id.value
+
+    @property
+    def number(self):
+        """
+        :type: integer
+        """
+        self._completeIfNotSet(self._number)
+        return self._number.value
+
+    @property
+    def owner_url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._owner_url)
+        return self._owner_url.value
+
+    @property
+    def state(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._state)
+        return self._state.value
+
+    @property
+    def updated_at(self):
+        """
+        :type: datetime.datetime
+        """
+        self._completeIfNotSet(self._updated_at)
+        return self._updated_at.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        self._completeIfNotSet(self._url)
+        return self._url.value
+
+    def get_columns(self):
+        """
+        :calls: `GET /projects/:project_id/columns <https://developer.github.com/v3/projects/columns/#list-project-columns>`_
+        :rtype: :class:`github.PaginatedList.PaginatedList` of :class:`github.ProjectColumn.ProjectColumn`
+        """
+
+        return github.PaginatedList.PaginatedList(
+            github.ProjectColumn.ProjectColumn,
+            self._requester,
+            self.columns_url,
+            None,
+            PROJECT_PREVIEW_HEADERS
+        )
+
+    def _initAttributes(self):
+        self._body = github.GithubObject.NotSet
+        self._columns_url = github.GithubObject.NotSet
+        self._created_at = github.GithubObject.NotSet
+        self._creator = github.GithubObject.NotSet
+        self._html_url = github.GithubObject.NotSet
+        self._id = github.GithubObject.NotSet
+        self._name = github.GithubObject.NotSet
+        self._node_id = github.GithubObject.NotSet
+        self._number = github.GithubObject.NotSet
+        self._owner_url = github.GithubObject.NotSet
+        self._state = github.GithubObject.NotSet
+        self._updated_at = github.GithubObject.NotSet
+        self._url = github.GithubObject.NotSet
+
+    def _useAttributes(self, attributes):
+        if "body" in attributes:  # pragma no branch
+            self._body = self._makeStringAttribute(attributes["body"])
+        if "columns_url" in attributes:  # pragma no branch
+            assert attributes["columns_url"] is None or isinstance(attributes["columns_url"], (str, unicode)), attributes["columns_url"]
```

Comment:
I believe the underlying `_makeXXXAttribute` method have None and type checks, so the additional assert is not needed

Reply:
Ah, ok - I'll double-check the behaviour of makeXXXAttribute and change if that's the case.